backend/app/Http/Controllers/AuthController.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\UpdatePasswordRequest;
+use App\Http\Resources\UserResource;
+use App\Models\User;
+
+class AuthController extends Controller
+{
+    //TODO: add login, signup, reset_password to the controller
+
+    public function update(UpdatePasswordRequest $request, $user_id)
```

Comment:
cast to user instance immediately 
```suggestion
    public function update(UpdatePasswordRequest $request, User $user)
```